### DISCUSSION TITLE
PDFViewer in Preact

### DISCUSSION THREAD

**Discussion by Johoseph** [diegomura/react-pdf#1483] [Q&A]
Hi all,

I was wanting to use this library in Preact and was encountering an error with the PDFViewer component freezing the browser on attempted render. The other components are working correctly (Document, Page, etc.) so I presume the issue has something to do with how I am aliasing react to preact. For reference, I have included my webpack and preact config files below:

### webpack config
```
import webpack from "webpack";

module.exports = {
  resolve: {
    alias: {
      react: "preact/compat",
      "react-dom": "preact/compat",
    },
    fallback: {
      process: "process/browser",
      stream: "stream-browserify",
      zlib: "browserify-zlib",
    },
    plugins: [
      new webpack.ProvidePlugin({
        process: "process/browser",
        Buffer: ["buffer", "Buffer"],
      }),
    ],
  },
};
```

### preact config
```
import path from "path";
import envVars from "preact-cli-plugin-env-vars";

export default function (config, env, helpers) {
  // absolute import logic lives here, removed for 

  envVars(config, env, helpers);
  config.node.process = true;
  config.node.Buffer = true;
}

```

Has anyone had any experience with this or would have any starting points on debugging the frozen PDFViewer?

Thanks!

**Comment by diegomura** [comment]
Hey @Johoseph . To be honest I never did something like this not used preact in the past. I'll leave it open in case someone has a better answer than mine 😄

**Comment by aymanxdev** [comment]
Hi @diegomura, is it possible to use the library without modify the webpack.config file? 
I was excited to see V2 released but still requires ejecting the script files and modifying them. Can we find a solution for this in order to have all the necessary things out of the box. Thanks!
